Lib/test/test_tokenize.py

```diff
@@ -15,23 +16,34 @@ class TokenizeTest(TestCase):
     # Tests for the tokenize module.
 
     # The tests can be really simple. Given a small fragment of source
-    # code, print out a table with tokens. The ENDMARKER is omitted for
-    # brevity.
+    # code, print out a table with tokens. The ENDMARKER, ENCODING and
+    # final NEWLINE are omitted for brevity.
 
     def check_tokenize(self, s, expected):
         # Format the tokens in s in a table format.
-        # The ENDMARKER is omitted.
+        # The ENDMARKER and final NEWLINE are omitted.
         result = []
         f = BytesIO(s.encode('utf-8'))
+        num_lines = len(s.splitlines())
         for type, token, start, end, line in tokenize(f.readline):
             if type == ENDMARKER:
                 break
+            if s[-1] not in '\r\n' and type == NEWLINE and end[0] == num_lines:
```

Comment:
This check is quite unreadable. Not simple to make more readable though. A short comment could help.